merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/services/ConstraintAction.java

```diff
@@ -120,145 +121,126 @@ public Map<Constraint, Fallible<?>> getViolations(final PlanId planId, final Opt
-                                                              + planId.id()));
 
-            for (final var _entry : ProfileSet.unwrapOptional(newProfiles.realProfiles()).entrySet()) {
-              if (!realProfiles.containsKey(_entry.getKey())) {
-                realProfiles.put(_entry.getKey(), LinearProfile.fromSimulatedProfile(_entry.getValue().segments()));
-              }
+      //compile
+      final var compiledConstraints = new ArrayList<ExecutableConstraint>();
+      for (final var constraint : constraintCode) {
+        switch (constraint.type()) {
+          case ConstraintType.EDSL e -> {
+            final var compilationResult = tryCompileEDSLConstraint(
+                plan.missionModelId(),
+                planId,
+                simDatasetId,
+                constraint);
+
+            if (compilationResult.isFailure()) {
+              final Fallible<ConstraintResult, List<? extends Exception>> r = Fallible.failure(compilationResult.getFailure().errors(), compilationResult.getMessage());
+              constraintResultMap.put(constraint, r);
+              continue;
             }
 
-            for (final var _entry : ProfileSet.unwrapOptional(newProfiles.discreteProfiles()).entrySet()) {
-              if (!discreteProfiles.containsKey(_entry.getKey())) {
-                discreteProfiles.put(
-                    _entry.getKey(),
-                    DiscreteProfile.fromSimulatedProfile(_entry.getValue().segments()));
-              }
-            }
-          } catch (InputMismatchException ex) {
-            constraintResultMap.put(constraint, Fallible.failure(ex));
-            continue;
+            compiledConstraints.add(new ExecutableConstraint.EDSLConstraint(constraint, compilationResult.get()));
           }
+          case ConstraintType.JAR j -> compiledConstraints.add(new ExecutableConstraint.JARConstraint(constraint));
         }
+      }
 
-        final Interval bounds = Interval.between(Duration.ZERO, simDuration);
-        final var preparedResults = new gov.nasa.jpl.aerie.constraints.model.SimulationResults(
-            simStartTime,
-            bounds,
-            activities,
-            realProfiles,
-            discreteProfiles);
-
-        ConstraintResult constraintResult = expression.evaluate(preparedResults, environment);
+      // sort constraints
+      Collections.sort(compiledConstraints);
 
-        constraintResult.constraintName = entry.getValue().name();
-        constraintResult.constraintRevision = entry.getValue().revision();
-        constraintResult.constraintId = entry.getKey();
-        constraintResult.resourceIds = List.copyOf(names);
+      // prepare simulation results -- all resources need to be fetched ahead of time as it is unknown what profiles
+      //    a procedural constraint will access
+      final var merlinResults = resultsHandle.getSimulationResults();
+      final var constraintsResults = new SimulationResults(merlinResults);
```

Comment:
On my first read through, `constraintResults` evoked the result of running a constraint, so I was surprised to discover that it's referring to simulation results in the format expected by constraints. `constraintResultMap` is similarly named, but means something different

Reply:
I've renamed `merlinResults` to `merlinSimResults` and `constraintsResults` to `edslSimResults`.